pycalphad/mapping/strategy/binary_strategy.py

```diff
@@ -0,0 +1,223 @@
+
+    # Sort axis by state variables first
+    # This shouldn't be necessary since dT/dX return np.nan, which we replace to np.inf for getting the optimal direction
+    # But just in case, we default to stepping in temperature if no direction could be found
+    axis_vars = map_utils._sort_axis_by_state_vars(axis_vars)
+
+    # Remove first axis variable from point - assumes we default to stepping along second variable
+    # We keep a copy of the conditions in case we decide that stepping along the first variable is better
+    comp_sets = point.stable_composition_sets
+
+    # 4 options here as a combination of
+    #    Fix CS 1 and free CS 2, or fix CS 2 and free CS 1
+    #    d(av1)/d(av2) or d(av2)/d(av1)
+    cs_options = [[comp_sets[0], comp_sets[1]], [comp_sets[1], comp_sets[0]]]
+    av_options = [[axis_vars[0], axis_vars[1]], [axis_vars[1], axis_vars[0]]]
+    options_tests = []
+    for options in itertools.product(cs_options, av_options):
+        cs_list, av_list = options
+        new_point = map_utils._generate_point_with_fixed_cs(point, cs_list[0], cs_list[1])
+        der = abs(zeq.compute_derivative(new_point, av_list[0], av_list[1]))
+        if np.isnan(der):
+            der = np.inf
+        else:
+            # Normalize derivative (normalization factor should be is axis delta)
+            der *= norm[av_list[1]] / norm[av_list[0]]
+        options_tests.append((der, new_point, av_list[0]))
+
+    # Best point/axis var is determined by the lowest derivative over 1
+    best_index = -1
+    best_der = np.inf
+    for i in range(len(options_tests)):
+        if options_tests[i][0] > 1:
+            _log.info(f"Option: Axis var {options_tests[i][2]}, derivative {options_tests[i][0]}, point {options_tests[i][1].fixed_phases}, {options_tests[i][1].free_phases}")
+            if options_tests[i][0] <= best_der:
+                best_index = i
+                best_der = options_tests[i][0]
+
+    # If no best point/axis var, then use the first one (which will be along a state variable)
+    if best_index == -1:
+        return options_tests[0][1], options_tests[0][2]
+    else:
+        return options_tests[best_index][1], options_tests[best_index][2]
+
+class BinaryStrategy(MapStrategy):
+    def __init__(self, dbf: Database, components: list[str], phases: list[str], conditions: dict[v.StateVariable, Union[float, tuple[float]]], **kwargs):
+        super().__init__(dbf, components, phases, conditions, **kwargs)
+
+    def initialize(self):
+        """
+        Searches axis limits to find starting points
+
+        Here, we do a step mapping along the axis bounds and grab all the nodes
+        The nodes of a step map is distinguished from starting points in that they have a parent
+
+        TODO: we should check performance to see if step mapping versus running multiple equilibrium along the axis then searching through the array is faster
```

Comment:
Resolve TODO

Reply:
I think this TODO should be resolved. From the SGTE binaries, it seems like most binaries are finishing under a minute.